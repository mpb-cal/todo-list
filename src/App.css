
button[disabled]:hover {
  cursor: default;
}

.new {
  background-color: aquamarine;
}

.justMoved {
  background-color: lightgreen;
}

.actionItemRow:hover {
  background-color: #ddd;
}
