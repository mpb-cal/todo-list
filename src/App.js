import React from 'react';
import { createStore, combineReducers } from 'redux';
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';
//import { C, items } from './actions';
import { items } from './actions';
import sampleState from './sampleStateMPB.json';
import ActionItemManagerRedux from './action-item-manager-redux';

//const STORAGE_KEY = 'action-item-manager-state';
const STORAGE_KEY_REDUX = 'action-item-manager-state-redux';

/*
const saveState = (userData) => {
  localStorage.setItem(STORAGE_KEY, JSON.stringify(userData));
}
*/

function App() {
  console.log(localStorage);

  const store = createStore(
    combineReducers({items}),
      JSON.parse(localStorage.getItem(STORAGE_KEY_REDUX)) || 
      sampleState
  );

  store.subscribe(() => {
    localStorage.setItem(STORAGE_KEY_REDUX, JSON.stringify(store.getState()));
  });

  return (
    <ActionItemManagerRedux store={store} />
  );
}

export default App;
