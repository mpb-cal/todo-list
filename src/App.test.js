import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing (shallow test)', () => {
  shallow(<App />);
});

it('renders welcome message', () => {
  const { getByText } = render(<App />);
  expect(getByText('Learn React')).toBeInTheDocument();
});