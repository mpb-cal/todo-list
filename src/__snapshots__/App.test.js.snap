// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="justify-content-center row"
    >
      <div
        class="col-auto"
      >
        <h1
          class="text-center"
        >
          TODO List
        </h1>
      </div>
    </div>
    <div
      class="d-print-none row"
    >
      <div
        class="col"
      >
        <button
          class="mb-3 btn btn-primary"
          type="button"
        >
          Export JSON
        </button>
        <form
          class=""
        >
          <div
            class="form-row"
          >
            <div
              class="form-group"
            >
              <div
                class="col"
              >
                Import JSON:
              </div>
              <div
                class="col"
              >
                <input
                  class="mb-3 form-control-file"
                  type="file"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <form
      class=""
    >
      <div
        class="d-print-none form-row"
      >
        <div
          class="col-10"
        >
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="newItem"
            />
          </div>
        </div>
        <div
          class="col-2"
        >
          <button
            class="btn btn-primary"
            type="submit"
          >
            Add Item
          </button>
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_0"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_0"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          use separate columns for table and id
        </div>
        <div
          class="d-none d-print-block description col"
        >
          use separate columns for table and id
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_1"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_1"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          eventually delete all history and comments columns
        </div>
        <div
          class="d-none d-print-block description col"
        >
          eventually delete all history and comments columns
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_2"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_2"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          AFI reports for Leif
        </div>
        <div
          class="d-none d-print-block description col"
        >
          AFI reports for Leif
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_3"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_3"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          add field for ticket number, link to ticket page
        </div>
        <div
          class="d-none d-print-block description col"
        >
          add field for ticket number, link to ticket page
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_4"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_4"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          add field for reported issue - see XLS for drop down options, ability to add new option
        </div>
        <div
          class="d-none d-print-block description col"
        >
          add field for reported issue - see XLS for drop down options, ability to add new option
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_5"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_5"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          button to create UPS return label
        </div>
        <div
          class="d-none d-print-block description col"
        >
          button to create UPS return label
        </div>
      </div>
      <div
        class="actionItemRow  form-row"
      >
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Up
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Down
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            type="button"
          >
            Top
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <button
            class="btn btn-success btn-sm"
            disabled=""
            type="button"
          >
            Delete
          </button>
        </div>
        <div
          class="d-print-none col"
        >
          <div
            class="done form-group"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="done_6"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="done_6"
                title=""
              >
                Done
              </label>
            </div>
          </div>
        </div>
        <div
          class="d-print-none description col-md-8 col-6"
        >
          button to create UPS shipment
        </div>
        <div
          class="d-none d-print-block description col"
        >
          button to create UPS shipment
        </div>
      </div>
    </form>
  </div>
</div>
`;
